account-kit/smart-contracts/src/ma-v2/account/modularAccountV2.ts

```diff
@@ -110,12 +143,64 @@ export async function createModularAccountV2(
     deferredAction,
   } = config;
 
+  const signer = "signer" in config ? config.signer : undefined;
+
+  const credential = "credential" in config ? config.credential : undefined;
+
+  const getFn = "getFn" in config ? config.getFn : undefined;
+
+  const rpId = "rpId" in config ? config.rpId : undefined;
+
+  const client = createBundlerClient({
+    transport,
+    chain,
+  });
+
   const accountFunctions = await (async () => {
     switch (config.mode) {
+      case "webauthn": {
+        if (!credential)
+          throw new Error("Missing params for MAV2 webauthn mode");
+        const publicKey = credential.publicKey;
+        const { x, y } = parsePublicKey(publicKey);
+        const {
+          salt = 0n,
+          factoryAddress = getDefaultMAV2FactoryAddress(chain),
+          initCode,
+        } = config;
+
+        const getAccountInitCode = async () => {
+          if (initCode) {
+            return initCode;
+          }
+
+          return concatHex([
+            factoryAddress,
+            encodeFunctionData({
+              abi: accountFactoryAbi,
+              functionName: "createWebAuthnAccount",
+              args: [x, y, salt, entityId],
+            }),
+          ]);
+        };
+
+        const accountAddress = await getAccountAddress({
+          client,
+          entryPoint,
+          accountAddress: _accountAddress,
+          getAccountInitCode,
+        });
```

Comment:
@adamegyed did some work in some of the other account types to do this locally rather than having to make an eth call to the entrtypoint, we should do the same here